File: backend/app/schemas/points_rule.py
from pydantic import BaseModel


class PointsRuleCreate(BaseModel):
    rule_type: str
    event_level: str | None = None
    group_name: str | None = None
    result_type: str | None = None
    points: int


class PointsRuleUpdate(BaseModel):
    rule_type: str | None = None
    event_level: str | None = None
    group_name: str | None = None
    result_type: str | None = None
    points: int | None = None
    enabled: bool | None = None


class PointsRuleResponse(BaseModel):
    id: int
    season_id: int
    rule_type: str
    event_level: str | None = None
    group_name: str | None = None
    result_type: str | None = None
    points: int
    enabled: bool

File: frontend/shared/types/pointsRule.ts
import type { RuleType, TournamentLevel, ResultType } from './enums'

export interface PointsRule {
  id: number
  season_id: number
  rule_type: RuleType
  event_level?: TournamentLevel
  group_name?: string
  result_type?: ResultType
  points: number
  description?: string
  created_at: string
  updated_at: string
}

export interface PointsRuleCreate {
  rule_type: RuleType
  event_level?: TournamentLevel
  group_name?: string
  result_type?: ResultType
  points: number
  description?: string
}

export interface PointsRuleUpdate {
  rule_type?: RuleType
  event_level?: TournamentLevel
  group_name?: string
  result_type?: ResultType
  points?: number
  description?: string
}
